src/SWP/Component/TemplatesSystem/Gimme/Widget/AbstractWidgetHandler.php

```diff
@@ -88,9 +88,12 @@ public function isVisible()
     /**
      * Render given template with given parameters.
      */
-    protected function renderTemplate($templateName, $parameters = array())
+    protected function renderTemplate($templateName, $parameters = null)
     {
-        $this->container->get('templating')->render(self::WIDGET_TEMPLATE_PATH.'/'.$templateName, $parameters);
+        if (null === $parameters) {
+            $parameters = $this->getAllParametersWithValue();
```

Comment:
getModelParameter now returns null, if the value is not set and there is no default value, although I think that in this case we should throw an exception as there will otherwise probably be an error in the template, although twig, unlike smarty, may actually report the name of the variable which hasn't been set.